storage/cloud-client/iam_test.py

```diff
@@ -22,24 +22,27 @@
 import storage_add_bucket_iam_member
 import storage_add_bucket_conditional_iam_binding
 import storage_view_bucket_iam_members
+import storage_remove_bucket_conditional_iam_binding
 
 MEMBER = "group:dpebot@google.com"
 ROLE = "roles/storage.legacyBucketReader"
 
 CONDITION_TITLE = "match-prefix"
 CONDITION_DESCRIPTION = "Applies to objects matching a prefix"
-CONDITION_EXPRESSION = "resource.name.startsWith(\"projects/_/buckets/bucket-name/objects/prefix-a-\")"
+CONDITION_EXPRESSION = (
+    'resource.name.startsWith("projects/_/buckets/bucket-name/objects/prefix-a-")'
+)
 
 
 @pytest.fixture
 def bucket():
     bucket = None
-    while bucket is None or bucket.exists():
+    while bucket is None or not bucket.exists():
+        storage_client = storage.Client()
         bucket_name = "test-iam-{}".format(uuid.uuid4())
-        bucket = storage.Client().bucket(bucket_name)
-    bucket.create()
-    bucket.iam_configuration.uniform_bucket_level_access_enabled = True
-    bucket.patch()
+        bucket = storage_client.bucket(bucket_name)
+        bucket.iam_configuration.uniform_bucket_level_access_enabled = True
```

Comment:
This shouldn't be indented. There is a possible bucket collision and this may be manipulating that instead.

At minimum the creation needs to be outside of the while loop.